File: adlatus-gui/libs/business/src/lib/agreement-management/agreement.service.ts
import {Inject, Injectable} from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Agreement, AgreementCreate, AgreementUpdate} from "@adlatus-gui/domain/agreement-management";

@Injectable({
  providedIn: 'root'
})
export class AgreementService {

  constructor(public httpClient: HttpClient, @Inject('environment') private environment: Record<string, string | boolean | number>) {
  }

  createAgreement(agreementCreate: AgreementCreate) {
    return this.httpClient.post<Agreement>((<string>this.environment.ENDPOINT), agreementCreate);
  }

  retrieveAllAgreements() {
    return this.httpClient.get<Agreement[]>((<string>this.environment.ENDPOINT));
  }

  retrieveAgreement(id: string) {
    return this.httpClient.get<Agreement>(`${(<string>this.environment.ENDPOINT)}/${id}`);
  }

  patchAgreement(id: string, agreementUpdate: AgreementUpdate) {
    return this.httpClient.patch<Agreement>(`${(<string>this.environment.ENDPOINT)}/${id}`, agreementUpdate)
  }

  deleteAgreement(id: string) {
    return this.httpClient.delete<void>(`${(<string>this.environment.ENDPOINT)}/${id}`);
  }
}

File: adlatus-agreement-api/src/main/java/de/bitconex/agreement/service/AgreementService.java
package de.bitconex.agreement.service;

import de.bitconex.agreement.model.Agreement;
import de.bitconex.agreement.model.AgreementCreate;
import de.bitconex.agreement.model.AgreementUpdate;

import java.util.List;

public interface AgreementService {

    List<Agreement> getAllAgreements();

    Agreement getAgreement(String agreementId);

    Agreement createAgreement(AgreementCreate agreementCreate);

    void deleteAgreement(String agreementId);

    Agreement patchAgreement(String agrementId, AgreementUpdate agreementUpdate);
}
